specs-go/v1/config.go

```diff
@@ -75,6 +75,7 @@ type History struct {
 }
 
 // Image is the JSON structure which describes some basic information about the image.
+// This provides the `application/vnd.oci.image.config.v1+json` mediatype when marshalled to JSON.
```

Comment:
@wking You're arguing that there's dichotomy between "free for all, no need to have `mediaType`" and "hueristic so that we can recognise every blob type". I don't think there is one.

The benefit of being able to know *what a JSON blob is meant to represent* is entirely separate from "I can tell what every blob in the image is without references". If you're not happy with detecting `tar` files (like `file` and `libmagic` do) that's fine. But please let's not make all of our JSON objects meaningless blobs that require jumping through references in order to even understand what we're looking at (or keeping the type information out-of-band).